project/app/views.py

```diff
@@ -72,20 +114,64 @@ def post(self, request):
 
 
 class AdminAccessView(ListAPIView, CreateAPIView):
-    permission_classes = [IsAuthenticated,RequiredAdmin]
+    permission_classes = [IsAuthenticated, RequiredAdmin]
 
     queryset = UserProfile.objects.all()
     serializer_class = UserProfileSerializer
 
 
 class UpdateProfileView(APIView):
-    permission_classes = [IsAuthenticated,RequiredAdmin]
+    permission_classes = [IsAuthenticated, RequiredAdmin]
 
-    def put(self, request,pk):
+    def put(self, request, pk):
         user_profile = UserProfile.objects.get(user_id=pk)
         profile_serializer = UserProfileSerializer(instance=user_profile, data=request.data)
 
         if not profile_serializer.is_valid():
             return Response(status=status.HTTP_404_NOT_FOUND)
         profile_serializer.save()
         return Response(profile_serializer.data)
+
+
+class AdminAddUserView(APIView):
+    permission_classes = [IsAuthenticated, RequiredAdmin, AdminCanCreate]
```

Comment:
We don't need different decorator, The RequiredAdmin and AdminCanCreate will do the same thing.

So, RequiredAdmin will be enough